crates/ra_syntax/src/grammar/expressions.rs

```diff
@@ -42,6 +42,11 @@ pub(crate) fn block(p: &mut Parser) {
     }
     let m = p.start();
     p.bump();
+    // test nocontentexpr
+    // fn foo(){
+    //     ;;;some_expr();;;;{;;;};;;;Ok(())
+    // }
+    while p.eat(SEMI) {}
```

Comment:
Hm, can we perhaps have only one `while p.eat(SEMI)` loop? I think if you put it right before `match` it'll do the trick? Or maybe even just add `SEMI => p.bump(),` case to match itself?

Reply:
I fixed a couple of bugs, which got rid of error nodes. That could feasibly be the reason